### PR TITLE
Add fetch method to Drops

### LINKED ISSUE #4615: New Drops::DocumentDrop page object doesn't support .fetch() method
###### What version of Jekyll are you using (`jekyll -v`)?
3.1.2

###### What operating system are you using?
CentOS 7.2

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, and the command you ran)
Executing a plugin function that looks up dotted names and returns jekyll hash values, i.e., "site.baseurl", "page.url", etc.  Recursively does this function, i.e.,:

 def lookup(context, name)                                                                                                    
 31      lookup = context                                                                                                           
 32      name.split(".").each { |value| lookup = lookup[value] }                                                                    
 33      lookup                                                                                                                     
 34    end           

###### What did you expect to see?

The value of context.registers[:page]['url']

###### What did you see instead?
NoMethodError: undefined method `fetch' for #<Jekyll::Drops::DocumentDrop:0x00000003390410>

The issue is that there seems to be a new object type, Jekyll::Drops:DocumentDrop introduced in commit 82c3ee365f5a8b948ce2ea853d75a954fa709939, full text:

   Initial work on using Liquid::Drops instead of Hashes.  …
   parkr committed on Dec 21, 2015
   The properties of Liquid::Drops are only evaluated when they're asked for
   and therefore save computation time. This prevents a lot of GC time cleaning
   up objects that are not needed, because they're not created unless requested.
   Additionally, this saves time for actual computation of those values because
   they can be computed only if needed.
   It's funny how much it helps when you only do what is needed. Far less overhead.

The new object doesn't support the standard '.fetch()' method on hashes.  As such, it behaves differently than other non-page objects ... 

i.e:

ls lookup
Liquid::Drop#methods: before_method  context=  has_key?  invoke_drop  to_liquid
Jekyll::Drops::Drop#methods: []  []=  content_methods  each_key  inspect  key?  keys  merge  merge!  to_h  to_hash
Jekyll::Drops::DocumentDrop#methods: 
  collection  content  excerpt  fallback_data  id  next  output  path  previous  relative_path  to_s  url
instance variables: @content_methods  @context  @mutations  @obj

and a standard hash:
ls {}
Enumerable#methods: 
  all?            count       each_cons         entries     flat_map  map     minmax     reduce        take      
  any?            cycle       each_entry        find        grep      max     minmax_by  reverse_each  take_while
  chunk           detect      each_slice        find_all    group_by  max_by  none?      slice_before  to_set    
  collect         drop        each_with_index   find_index  inject    min     one?       sort          zip       
  collect_concat  drop_while  each_with_object  first       lazy      min_by  partition  sort_by     
JSON::Ext::Generator::GeneratorMethods::Hash#methods: to_json
Hash#methods: 
  ==                    default        each_key    has_key?    keep_if  merge!              replace  to_h       values_at
  []                    default=       each_pair   has_value?  key      pretty_print        select   to_hash  
  []=                   default_proc   each_value  hash        key?     pretty_print_cycle  select!  to_liquid
  assoc                 default_proc=  empty?      include?    keys     rassoc              shift    to_s     
  clear                 delete         eql?        index       length   rehash              size     update   
  compare_by_identity   delete_if      fetch       inspect     member?  reject              store    value?   
  compare_by_identity?  each           flatten     invert      merge    reject!             to_a     values   

This breaks a *bunch* of code ... sure, I can work around it, but it seems that having to special case each object type complicates plugin code when in fact you are looking to greatly reduce the complications.  

If there are better, less fragile ways to do the lookup please let me know.
Thanks for any feedback ... good stuff.

-Fred Cassirer

### PR DESCRIPTION
**Pull request by ayastreb**
Fixes #4615 

As pointed out by @parkr - added `fetch` method to `Drops::Drop` class.

### CODE REVIEW

File: lib/jekyll/drops/drop.rb [hunk 1]
```diff
@@ -199,6 +199,10 @@ def merge!(other)
           end
         end
       end
+
+      def fetch(*args, &block)
+        to_h.fetch(*args, &block)
```
Comment (parkr) [inline lib/jekyll/drops/drop.rb]: `#to_h` is an expensive operation. Could we mimic the behaviour without having to actually create a hash?

**Comment by parkr** [comment]
This LGTM. @jekyll/core?

**Comment by envygeeks** [comment]
:shipit:

**Comment by parkr** [comment]
Thanks @envygeeks :)

@jekyllbot: merge +minor

**Pull request merged**